diff -Naur corkscrew-2.0.orig/config.h corkscrew-2.0/config.h
--- corkscrew-2.0.orig/config.h	1970-01-01 01:00:00.000000000 +0100
+++ corkscrew-2.0/config.h	2010-03-10 15:20:50.681040766 +0100
@@ -0,0 +1,39 @@
+/* config.h.  Generated automatically by configure.  */
+/* config.in.  Generated automatically from configure.in by autoheader.  */
+
+/* Define to empty if the keyword does not work.  */
+/* #undef const */
+
+/* Define if you have the ANSI C header files.  */
+#define STDC_HEADERS 1
+
+/* Define if you can safely include both <sys/time.h> and <time.h>.  */
+#define TIME_WITH_SYS_TIME 1
+
+/* Define if you have the select function.  */
+#define HAVE_SELECT 1
+
+/* Define if you have the socket function.  */
+#define HAVE_SOCKET 1
+
+/* Define if you have the strerror function.  */
+#define HAVE_STRERROR 1
+
+/* Define if you have the <string.h> header file.  */
+/* #undef HAVE_STRING_H */
+
+/* Define if you have the <sys/filio.h> header file.  */
+/* #undef HAVE_SYS_FILIO_H */
+
+/* Define if you have the xnet library (-lxnet).  */
+/* #undef HAVE_LIBXNET */
+
+/* Name of package */
+#define PACKAGE "corkscrew"
+
+/* Version number of package */
+#define VERSION "2.0"
+
+/* Define if compiler has function prototypes */
+#define PROTOTYPES 1
+
